From f4a092c1510f325e7e158583226c479617fc5ab4 Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Fri, 8 Dec 2023 09:29:17 +0100
Subject: [PATCH] [freetype] Fix pkgconfig files to be relocatable

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd04b97..6087f42 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -561,7 +561,7 @@ if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
 
   string(REPLACE ";" ", " PKGCONFIG_REQUIRES_PRIVATE "${PKGCONFIG_REQUIRES_PRIVATE}")
 
-  string(REPLACE "%prefix%" ${CMAKE_INSTALL_PREFIX}
+  string(REPLACE "%prefix%" "\${pcfiledir}/../.."
           FREETYPE2_PC_IN ${FREETYPE2_PC_IN})
   string(REPLACE "%exec_prefix%" "\${prefix}"
           FREETYPE2_PC_IN ${FREETYPE2_PC_IN})
-- 
2.25.1

