diff --git a/src/win32/win32.c b/src/win32/win32.c
index 943e400..f3e7809 100644
--- a/src/win32/win32.c
+++ b/src/win32/win32.c
@@ -48,6 +48,7 @@ int usleep(unsigned int useconds)
 }
 
 
+#if !defined(_POSIX_TIMERS) || (_POSIX_TIMERS <= 0)
 int nanosleep( const struct timespec * rqtp, struct timespec * rmtp )
 {
 	if (rqtp->tv_nsec > 999999999) {
@@ -56,7 +57,8 @@ int nanosleep( const struct timespec * rqtp, struct timespec * rmtp )
 		return -1;
 	}
 	return usleep( rqtp->tv_sec * 1000000 + rqtp->tv_nsec / 1000 );
-} 
+}
+#endif
 
 int setenv(const char *name, const char *value, int overwrite)
 {