From 7a231d83a3462637a87f2e971073e7935c71b3a3 Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Tue, 14 Apr 2026 17:26:18 +0200
Subject: [PATCH] Fix distinfo.py to support --prefix option on Windows
 properly

---
 sipbuild/distinfo/distinfo.py | 39 ++++++++++++++++++++++++++---------
 1 file changed, 29 insertions(+), 10 deletions(-)

diff --git a/sipbuild/distinfo/distinfo.py b/sipbuild/distinfo/distinfo.py
index 2e67f090..656d88b0 100644
--- a/sipbuild/distinfo/distinfo.py
+++ b/sipbuild/distinfo/distinfo.py
@@ -55,6 +55,25 @@ def distinfo(name, console_scripts, gui_scripts, sbom_files, generator,
             generator_version=generator_version)
 
 
+def fix_abs_path(destdir, path):
+    #print("fix_abs_path: path={}, destdir={}".format(path, destdir))
+    if os.path.isabs(path):
+        if destdir:
+            #print("   fixing...")
+            normedPath = os.path.normpath(path)
+            normedPath = os.path.splitdrive(normedPath)[1]
+            if normedPath.startswith(os.sep):
+                normedPath = normedPath[1:]
+            outpath = os.path.join(destdir, normedPath)
+        else:
+            outpath = path
+    else:
+        raise UserException(
+            "fix_abs_path is provided with relative path'{}'".format(path))
+
+    #print("   result: {}".format(outpath))
+    return outpath
+
 def create_distinfo(distinfo_dir, wheel_tag, installed, metadata,
         requires_dists, project_root, console_scripts, gui_scripts, sbom_files,
         prefix_dir='', generator=None, generator_version=None):
@@ -67,7 +86,7 @@ def create_distinfo(distinfo_dir, wheel_tag, installed, metadata,
         generator_version = SIP_VERSION_STR
 
     # The prefix directory corresponds to DESTDIR or INSTALL_ROOT.
-    real_distinfo_dir = prefix_dir + distinfo_dir
+    real_distinfo_dir = fix_abs_path(prefix_dir, distinfo_dir)
 
     # Make sure we have an empty dist-info directory.  Handle exceptions as the
     # user may be trying something silly with a system directory.
@@ -107,7 +126,7 @@ def create_distinfo(distinfo_dir, wheel_tag, installed, metadata,
         installer_fn = os.path.join(distinfo_dir, 'INSTALLER')
         installed.append(installer_fn)
 
-        with open(prefix_dir + installer_fn, 'w') as installer_f:
+        with open(fix_abs_path(prefix_dir, installer_fn), 'w') as installer_f:
             print(generator, file=installer_f)
     else:
         # Define any entry points.
@@ -115,7 +134,7 @@ def create_distinfo(distinfo_dir, wheel_tag, installed, metadata,
             eps_fn = os.path.join(distinfo_dir, 'entry_points.txt')
             installed.append(eps_fn)
 
-            with open(prefix_dir + eps_fn, 'w') as eps_f:
+            with open(fix_abs_path(prefix_dir, eps_fn), 'w') as eps_f:
                 if console_scripts:
                     eps_f.write(
                             '[console_scripts]\n' + '\n'.join(
@@ -135,7 +154,7 @@ Tag: {}
         wheel_fn = os.path.join(distinfo_dir, 'WHEEL')
         installed.append(wheel_fn)
 
-        with open(prefix_dir + wheel_fn, 'w') as wheel_f:
+        with open(fix_abs_path(prefix_dir, wheel_fn), 'w') as wheel_f:
             wheel_f.write(
                     WHEEL.format(WHEEL_VERSION, generator, generator_version,
                             wheel_tag))
@@ -150,11 +169,11 @@ Tag: {}
     record_fn = os.path.join(distinfo_dir, 'RECORD')
 
     distinfo_path, distinfo_base = os.path.split(distinfo_dir)
-    real_distinfo_path = os.path.normcase(prefix_dir + distinfo_path)
+    real_distinfo_path = os.path.normcase(fix_abs_path(prefix_dir, distinfo_path))
 
-    with open(prefix_dir + record_fn, 'w') as record_f:
+    with open(fix_abs_path(prefix_dir, record_fn), 'w') as record_f:
         for name in installed:
-            real_name = prefix_dir + name
+            real_name = fix_abs_path(prefix_dir, name)
             if os.path.isdir(real_name):
                 all_fns = []
 
@@ -176,7 +195,7 @@ Tag: {}
 
                 if norm_fn.startswith(real_distinfo_path):
                     fn_name = fn[len(real_distinfo_path) + 1:].replace('\\', '/')
-                elif norm_fn.startswith(prefix_dir + sys.prefix):
+                elif norm_fn.startswith(fix_abs_path(prefix_dir, sys.prefix)):
                     fn_name = os.path.relpath(
                             fn, real_distinfo_path).replace('\\', '/')
                 else:
@@ -206,7 +225,7 @@ def write_metadata(metadata, requires_dists, metadata_fn, project_root,
 
         metadata['requires-dist'] = requires_dists + rd
 
-    with open(prefix_dir + metadata_fn, 'w') as metadata_f:
+    with open(fix_abs_path(prefix_dir, metadata_fn), 'w') as metadata_f:
         description = None
 
         # Do these first for cosmetic reasons.
@@ -240,7 +259,7 @@ def _install_files(files, prefix_dir, target_dir, installed):
                 file_fn = os.path.join(target_dir, file)
                 installed.append(file_fn)
 
-                real_dir = prefix_dir + os.path.dirname(file_fn)
+                real_dir = fix_abs_path(prefix_dir, os.path.dirname(file_fn))
                 os.makedirs(real_dir, exist_ok=True)
                 shutil.copy(file, real_dir)
 
-- 
2.49.0.windows.1

