diff --git a/meson.build b/meson.build
index a1d04f876..fb2e9964f 100644
--- a/meson.build
+++ b/meson.build
@@ -1570,13 +1570,13 @@ endif
 ### process_detect ###
 
 # MSVC requires declspec on data symbols when importing from a shared library.
-if features.get('shared', false) and not ['linux', 'solaris', 'darwin', 'ios', 'iphonesimulator'].contains(tgt_os) 
-	if features.get('gcc', false) # Looks like a typo on upstream
-		warning('Shared library build is only supported on ELF; assuming this is OK')
-	else
- 		error('Shared library build is only supported on ELF and Darwin for now')
-	endif
-endif
+# if features.get('shared', false) and not ['linux', 'solaris', 'darwin', 'ios', 'iphonesimulator'].contains(tgt_os)
+# 	if features.get('gcc', false) # Looks like a typo on upstream
+# 		warning('Shared library build is only supported on ELF; assuming this is OK')
+# 	else
+#  		error('Shared library build is only supported on ELF and Darwin for now')
+# 	endif
+# endif
 
 features.set('unistd_h', c.has_header('unistd.h'))
 if c.has_header('vpx/vpx_integer.h', include_directories: include_directories('.'))

