From c422d0c105e24f98d196399a52c8ce5c4cee6d82 Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Thu, 26 Sep 2024 15:33:13 +0200
Subject: [PATCH] Add debugging for the linker

---
 meson.build | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/meson.build b/meson.build
index b5cbd9a8b3..ed498a018f 100644
--- a/meson.build
+++ b/meson.build
@@ -2739,6 +2739,11 @@ foreach built_lib : built_libs
     capture: true
   )
 
+  message('Declaring a library: @0@'.format(built_lib[0]))
+  message('    deps: @0@'.format(deps))
+  message('    link_with: @0@'.format(link_with))
+  message('    link_args: @0@'.format(link_args))
+
   lib = library(built_lib[0],
     dependencies: deps,
     soversion: version,
-- 
2.34.1

