From 8c1811bcda9170beb0510cfab7d5d31b1413c63d Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Mon, 14 Apr 2025 13:06:13 +0200
Subject: [PATCH 03/10] Be more forceful when setting PIC on x86_32

---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index 9507aa6b2e..4588c8e509 100644
--- a/meson.build
+++ b/meson.build
@@ -1935,6 +1935,7 @@ if cpu == 'x86'
   conf.set('i686', 1)
   conf.set('x86_32', 1)
   conf.set('fast_cmov', 1)
+  conf.set('pic', 1)
 elif cpu == 'x86_64'
   arch = 'x86'
   conf.set('x86', 1)
-- 
2.54.0

