From c5d6b9ef96d787fb5042f6e0b6633aa7c5992c2b Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Tue, 15 Apr 2025 11:17:44 +0200
Subject: [PATCH 06/10] Fix building when iconv feature is explicitly disabled

---
 meson.build | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/meson.build b/meson.build
index 95d7f3fb46..bc84d3351c 100644
--- a/meson.build
+++ b/meson.build
@@ -2061,6 +2061,11 @@ foreach check : all_checks
   name = check[0]
   opt_name = options_map.get(name, name)
 
+  if (name == 'iconv' and get_option(opt_name).disabled())
+    message('Skipping iconv because disabled by the user')
+    continue
+  endif
+
   if check_type == 'func'
     headers = []
   elif check_type == 'header'
-- 
2.54.0

